crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -639,12 +647,12 @@ impl<'w, 's> SystemParamFetch<'w, 's> for WorldState {
 /// // .add_system(reset_to_system(my_config))
 /// # assert_is_system(reset_to_system(Config(10)));
 /// ```
-pub struct Local<'a, T: Resource>(&'a mut T);
+pub struct Local<'a, T: Send + Sync + 'static>(&'a mut T);
```

Comment:
I don't think this `Sync` bound should be needed, although if removing it doesn't work that's also fine.

(I wonder if there's an abstraction to turn `!Sync` into `Sync` by enforcing access through a unique reference)

Reply:
I agree here. Started with the trivial conversion though :)

Reply:
> (I wonder if there's an abstraction to turn `!Sync` into `Sync` by enforcing access through a unique reference)

There's [sync_wrapper](https://crates.io/crates/sync_wrapper) for exactly this purpose. A similar thing is [proposed](https://github.com/rust-lang/rust/pull/97629) in the standard library.